### PR TITLE
Infeng-6845: Add Frugal examples/skynet tests for activemq (stomp) transport

### PR DESCRIPTION
**Pull request by yuanmwang-wf**
### Story:
Add Frugal examples/skynet tests for activemq (stomp) transport

### My Test Results:
Tested examples work locally (need to change glide.yaml to pin frugal to a commit hash that has stomp transports since the transports are not released yet).

Skynet passed.

#### Reviewers:
@Workiva/product2

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.
<details>
<summary>(2) Image validation issues were detected</summary>
<li>Repo does not appear to install package updates in deployable image</li>
<li>Application appears to run using root user.</li>
</details>

## Action Items
* Ensure [package updates](https://wiki.atl.workiva.net/display/SECURITY/Security+Requirements+for+Building+Containers#SecurityRequirementsforBuildingContainers-PackageUpdating) are being installed.
* [Specify a non-root USER](https://wiki.atl.workiva.net/display/SECURITY/Security+Requirements+for+Building+Containers#SecurityRequirementsforBuildingContainers-RunningwithNon-rootUser) in your deploy stage.
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/integration/go/common/client.go [hunk 1]
```diff
@@ -68,16 +67,21 @@ func StartClient(
 	*/
 	if transport == NatsName {
 		go func() {
+			fmt.Printf("starting nats pubsub")
 			<-pubSub
 
 			if err != nil {
 				panic(err)
 			}
+			var pfactory frugal.FPublisherTransportFactory
```
Comment (brianshannan-wf) [inline test/integration/go/common/client.go]: Why did you need to change these variable declarations?

File: test/integration/go/common/client.go [hunk 2]
```diff
@@ -68,16 +67,21 @@ func StartClient(
 	*/
 	if transport == NatsName {
 		go func() {
+			fmt.Printf("starting nats pubsub")
```
Comment (brianshannan-wf) [inline test/integration/go/common/client.go]: Should be removed. If we want more logging long term we should use a logging library

File: test/integration/go/common/publisher.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		return fmt.Errorf("Invalid protocol specified %s", protocol)
+	}
+
+	go func() {
+		<-pubSub
+		var pfactory frugal.FPublisherTransportFactory
+		var sfactory frugal.FSubscriberTransportFactory
+
+		stompConn := getStompConn()
+		pfactory = frugal.NewFStompPublisherTransportFactory(stompConn, 32*1024*1024, "")
```
Comment (brianshannan-wf) [inline test/integration/go/common/publisher.go]: We should check the transport name here and only create this transport if it's stomp, as in the future I'd like to move the nats pub/sub tests to work like these tests. https://github.com/Workiva/frugal/pull/1048/files#diff-d35520586447c592f6b306056eda657cR38 is an example

File: test/integration/go/common/server.go [hunk 4]
```diff
@@ -46,16 +46,20 @@ func StartServer(
...
 	var err error
 
 	/*
 		Subscriber for Pub/Sub tests
 		Subscribe to events, publish response upon receipt
 	*/
 	go func() {
-		pfactory := frugal.NewFNatsPublisherTransportFactory(conn)
-		sfactory := frugal.NewFNatsSubscriberTransportFactory(conn)
+		var pfactory frugal.FPublisherTransportFactory
```
Comment (brianshannan-wf) [inline test/integration/go/common/server.go]: Same comment about variable declaration changes

File: test/integration/go/common/server.go [hunk 5]
```diff
@@ -83,7 +87,9 @@ func StartServer(
 				panic(err)
 			}
 			// Explicitly flushing the publish to ensure it is sent before the main thread exits
-			conn.Flush()
+			if transport == NatsName {
```
Comment (brianshannan-wf) [inline test/integration/go/common/server.go]: Out of curiosity, why was this added?

File: test/integration/go/common/server.go [hunk 6]
```diff
@@ -111,16 +117,19 @@ func StartServer(
 		// Healthcheck used in the cross language runner to check for server availability
 		http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {})
 		go http.ListenAndServe(hostPort, nil)
+		if err := server.Serve(); err != nil {
```
Comment (brianshannan-wf) [inline test/integration/go/common/server.go]: Why move this here? It duplicates this code to multiple places

File: test/integration/go/common/subscriber.go [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+	default:
+		panic(fmt.Errorf("Invalid protocol specified %s", protocol))
+	}
+
+	go func() {
+		var pfactory frugal.FPublisherTransportFactory
+		var sfactory frugal.FSubscriberTransportFactory
+
+		conn := getStompConn()
+		pfactory = frugal.NewFStompPublisherTransportFactory(conn, 32*1024*1024, "")
```
Comment (brianshannan-wf) [inline test/integration/go/common/subscriber.go]: Same comment about transport name

File: test/integration/tests.json [hunk 8]
```diff
@@ -15,15 +15,30 @@
       "timeout": 8,
       "command": [
         "go/bin/testserver"
-      ]
+      ],
+      "transports": [ "nats", "http" ]
```
Comment (brianshannan-wf) [inline test/integration/tests.json]: Why move this from the rpcTransports key and duplicate it in the client and server keys?

Comment (yuanmwang-wf) [reply test/integration/tests.json]: It took me a few tries to get the tests to work, I thought I wasn't configuring the tests the right so I moved them around. Should be addressed now.

File: test/integration/go/common/subscriber.go (hunk 7 above)
Comment (yuanmwang-wf) [reply test/integration/go/common/subscriber.go]: I was trying to include activemq into the same nats pub/sub test, using different connections depending on the transport name. That's why these changes were made, I forgot to switch them back after I realized they should be in their own files.

File: test/integration/go/common/server.go (hunk 6 above)
Comment (yuanmwang-wf) [reply test/integration/go/common/server.go]: Addressed in e0f722f

File: test/integration/go/common/server.go (hunk 5 above)
Comment (yuanmwang-wf) [reply test/integration/go/common/server.go]: At one point it was possible for this function to get a stomp transport, forgot to remove it. Addressed in e0f722f

File: test/integration/go/common/server.go (hunk 4 above)
Comment (yuanmwang-wf) [reply test/integration/go/common/server.go]: Addressed in e0f722f

File: test/integration/go/common/publisher.go (hunk 3 above)
Comment (yuanmwang-wf) [reply test/integration/go/common/publisher.go]: Addressed in e0f722f

File: test/integration/go/common/client.go (hunk 2 above)
Comment (yuanmwang-wf) [reply test/integration/go/common/client.go]: Addressed in e0f722f

File: test/integration/go/common/client.go (hunk 1 above)
Comment (yuanmwang-wf) [reply test/integration/go/common/client.go]: Addressed in e0f722f

**Comment by yuanmwang-wf** [comment]
This should be ready for review again @Workiva/product2

File: test/integration/go/common/subscriber.go (hunk 7 above)
Comment (brianshannan-wf) [reply test/integration/go/common/subscriber.go]: Shouldn't we check the transport name here so we can use this for the nats subscriber in the future though?

Comment (yuanmwang-wf) [reply test/integration/go/common/subscriber.go]: Yup, it should check... Done in d88fff4

**Comment by brianshannan-wf** [comment]
+1

**Comment by blakeroberts-wk** [comment]
+1

**Comment by yuanmwang-wf** [comment]
@Workiva/release-management-p
